addons/website/static/src/js/menu/new_content.js

```diff
@@ -2,28 +2,72 @@ odoo.define('website.newMenu', function (require) {
 'use strict';
 
 var core = require('web.core');
+var Dialog = require('web.Dialog');
 var websiteNavbarData = require('website.navbar');
 var wUtils = require('website.utils');
 
 var qweb = core.qweb;
 var _t = core._t;
 
+var enableFlag = 'enable_new_content';
+
 var NewContentMenu = websiteNavbarData.WebsiteNavbarActionWidget.extend({
     xmlDependencies: ['/web_editor/static/src/xml/editor.xml'],
     actions: _.extend({}, websiteNavbarData.WebsiteNavbarActionWidget.prototype.actions || {}, {
         new_page: '_createNewPage',
     }),
     events: _.extend({}, websiteNavbarData.WebsiteNavbarActionWidget.prototype.events || {}, {
-        'click > a': '_onMenuToggleClick',
-        'click > #o_new_content_menu_choices': '_onBackgroundClick',
+        'click': '_onBackgroundClick',
+        'keydown': '_onBackgroundKeydown',
     }),
+    // allow text to be customized with inheritance
+    newContentText: {
+        failed: _t('"%s" failed to install.'),
+        installInProgress: _t('The installation of a module is already in progress.'),
+        installNeeded: _t('You need to install the "%s" module in order to execute this action.'),
+        installPleaseWait: _t('Please wait while "%s" is being installed...'),
+    },
 
     /**
+     * Prepare the navigation and find the modules to install.
+     * Move not installed module buttons after installed modules buttons,
+     * but keep the original index to be able to move back the pending install
+     * button at its final position, so the user can click at the same place.
+     *
      * @override
      */
     start: function () {
+        var self = this;
+        var autoOpen = typeof $.deparam.querystring()[enableFlag] !== 'undefined';
+        this.pendingInstall = 0;
         this.$newContentMenuChoices = this.$('#o_new_content_menu_choices');
-        return this._super.apply(this, arguments);
+
+        var $modules = this.$newContentMenuChoices.find('.js_new_element');
+        _.each($modules, function (el, index) {
+            var $el = $(el);
+            $el.data('original-index', index);
+            if ($el.data('module-id')) {
+
+                $el.appendTo($el.parent());
+                $el.find('a i').addClass('text-muted');
+                $el.find('a p').addClass('text-muted');
```

Comment:
I added the class on the i and p because the class needs to override custom css we put on them specifically. Should I try to improve the original css then?

Reply:
No, but at least do this:
`$el.find('a i, a p').addClass('text-muted')`

Reply:
ok